script/tool/lib/src/common/package_looping_command.dart

```diff
@@ -167,4 +222,86 @@ abstract class PackageLoopingCommand extends PluginCommand {
     }
     print(Colorize(heading)..cyan());
   }
+
+  /// Prints a summary of packges run, packages skipped, and warnings.
+  void _printRunSummary(
+      List<Directory> packages, Map<Directory, PackageResult> results) {
+    final Set<Directory> skippedPackages = results.entries
+        .where((MapEntry<Directory, PackageResult> entry) =>
+            entry.value.state == RunState.skipped)
+        .map((MapEntry<Directory, PackageResult> entry) => entry.key)
+        .toSet();
+    final int skipCount = skippedPackages.length;
+    // Split the warnings into those from packages that ran, and those that
+    // were skipped.
+    final Set<Directory> _skippedPackagesWithWarnings =
+        _packagesWithWarnings.intersection(skippedPackages);
+    final int skippedWarningCount = _skippedPackagesWithWarnings.length;
+    final int runWarningCount =
+        _packagesWithWarnings.length - skippedWarningCount;
+
+    final String runWarningSummary =
+        runWarningCount > 0 ? ' ($runWarningCount with warnings)' : '';
+    final String skippedWarningSummary =
+        runWarningCount > 0 ? ' ($skippedWarningCount with warnings)' : '';
+    print('------------------------------------------------------------');
+    if (hasLongOutput) {
+      _printPerPackageRunOverview(packages, skipped: skippedPackages);
+    }
+    print(
+        'Ran for ${packages.length - skipCount} package(s)$runWarningSummary');
+    if (skipCount > 0) {
+      print('Skipped $skipCount package(s)$skippedWarningSummary');
+    }
+    if (_otherWarningCount > 0) {
+      print('$_otherWarningCount warnings not associated with a package');
+    }
+  }
+
+  /// Prints a one-line-per-package overview of the run results for each
+  /// package.
+  void _printPerPackageRunOverview(List<Directory> packages,
+      {required Set<Directory> skipped}) {
+    print('Run overview:');
+    for (final Directory package in packages) {
+      final bool hadWarning = _packagesWithWarnings.contains(package);
+      Colorize summary;
+      if (skipped.contains(package)) {
+        if (hadWarning) {
+          summary = Colorize('skipped (with warning)')..lightYellow();
+        } else {
+          summary = Colorize('skipped')..darkGray();
+        }
+      } else {
+        if (hadWarning) {
+          summary = Colorize('ran (with warning)')..yellow();
+        } else {
+          summary = Colorize('ran')..green();
+        }
+      }
+      print('  ${getPackageDescription(package)} - $summary');
+    }
+    print('');
+  }
+
+  /// Prints a summary of all of the failures from [results].
+  void _printFailureSummary(
+      List<Directory> packages, Map<Directory, PackageResult> results) {
+    const String indentation = '  ';
+    printError(failureListHeader);
+    for (final Directory package in packages) {
+      final PackageResult result = results[package]!;
```

Comment:
nit: I don't think you need to pass in packages here, you can just loop over `results.values`

Reply:
Results is a map, and thus doesn't have a guaranteed order; it's important that the summary be in the same order as the packages for ease of referencing up to the details.